crates/driver/src/tests/cases/settle.rs

```diff
@@ -110,3 +111,50 @@ async fn high_gas_limit() {
         .unwrap();
     test.settle(&id).await.ok().await;
 }
+
+#[tokio::test]
+#[ignore]
+async fn too_many_settle_calls() {
+    let test = tests::setup()
+        .allow_multiple_solve_requests()
+        .pool(ab_pool())
+        .order(ab_order())
+        .solution(ab_solution())
+        .ethrpc_args(shared::ethrpc::Arguments {
+            ethrpc_max_batch_size: 10,
+            ethrpc_max_concurrent_requests: 10,
+            ethrpc_batch_delay: std::time::Duration::from_secs(1),
+        })
+        .solve_deadline_timeout(chrono::Duration::seconds(4))
+        .done()
+        .await;
+
+    let id1 = test.solve().await.ok().id();
+    let id2 = test.solve().await.ok().id();
+    let id3 = test.solve().await.ok().id();
+    let id4 = test.solve().await.ok().id();
```

Comment:
Due to the framework limitations, this has to be executed sequentially. Otherwise, each solution will most likely receive the same ID, which is critical for this test, where at least 4 distinct settlement requests must be sent.